src/pathwayvis/services/actions/base.ts

```diff
@@ -18,34 +18,31 @@ export abstract class Action {
 export abstract class ReactionAction extends Action {
   public shared: MapData;
 
-  public callback(ws: WSService, $timeout: angular.ITimeoutService, shared: SharedService): any {
-
+  public callback(
```

Comment:
@AlbaLopez I think this is the correct way to add the objective, you have to send it together with the rest of the data. But correct me if I'm wrong.

Reply:
Yes, I thought that I need to make a different call, but this is a better solution.